src/parameterizations/vertical/MOM_set_diffusivity.F90

```diff
@@ -337,6 +369,20 @@ subroutine set_diffusivity(u, v, h, u_h, v_h, tv, fluxes, optics, visc, dt, Kd_i
   if (CS%id_Kd_bkgnd > 0) allocate(dd%Kd_bkgnd(isd:ied,jsd:jed,nz+1), source=0.)
   if (CS%id_Kv_bkgnd > 0) allocate(dd%Kv_bkgnd(isd:ied,jsd:jed,nz+1), source=0.)
 
+  if (CS%id_kbbl > 0) allocate(dd%kbbl(isd:ied,jsd:jed), source=0.)
```

Comment:
It might be cleaner if we could figure out a way to move all of these diagnostics into the MOM_internal_tides module.  It should be noted, though, that such a refactoring would not change any answers, so it would not be a catastrophe if this PR were accepted in its current form and then the diagnostics were moved later.

Reply:
agreed to rework this in subsequent PR